python/ray/autoscaler/_private/util.py

```diff
@@ -103,38 +104,63 @@ def prepare_config(config):
     return with_defaults
 
 
-def rewrite_legacy_yaml_to_available_node_types(
-        config: Dict[str, Any]) -> Dict[str, Any]:
+def fillout_defaults(config: Dict[str, Any]) -> Dict[str, Any]:
+    defaults = _get_default_config(config["provider"])
+    defaults.update(config)
+    # Just for clarity:
+    merged_config = copy.deepcopy(defaults)
+    merged_config["auth"] = merged_config.get("auth", {})
```

Comment:
remove. why is this needed?

Reply:
.get("auth") was there before -- it could be avoided by some minor refactors elsewhere, but it should stay for now. I'll add a comment indicating what it's for.

The auth field is fed into NodeUpdater, so it needs to be there. On the other hand, it's not relevant for some node providers (like K8s) so it's currently absent from some configs.